### PR TITLE
Fix passing non-string variables for other bundle actions

### PR DESCRIPTION
**Pull request by carolynvs**
The install action had been updated in https://github.com/getporter/terraform-mixin/pull/91 to work with complex data types by using a json parser to read and then write out variables when converting them from porter.yaml values to -var flags.
This ensures that primitive data types are printed without unexpected formatting, such as quotes around a string value, while arrays and objects are printed in json.

This patch applies the same fix to the other bundle actions: upgrade, invoke and uninstall. I've also extracted that common logic into a function with tests so that we can be sure all the actions are using it correctly.

[Consolidate test bundle and example bundle](https://github.com/getporter/terraform-mixin/commit/a4976141643346f087823d21bb14bb75db0e598d)

The test bundle had a lot of great edge cases and examples for how to use terraform in a bundle that would be useful to make people aware of in our example bundle. I have updated the basic-tf-example bundle with the contents of our test bundle. I also added defaults for all the parameters so that it doesn't require any additional config to run.

This is also good because our example bundle is tested in CI now.

### CODE REVIEW

File: examples/basic-tf-example/porter.yaml [hunk 1]
```diff
@@ -50,21 +113,51 @@ show:
 plan:
   - terraform:
       description: "Invoke 'terraform plan'"
-      flags:
```
Comment (carolynvs) [inline examples/basic-tf-example/porter.yaml:53]: I removed these because they aren't helpful in an example, we don't actually want to suppress plan output or turn off color.
